Library/Homebrew/services/cli.rb

```diff
@@ -196,18 +196,27 @@ def self.stop(targets, verbose: false, no_wait: false, max_wait: 0, keep: false)
               System::Systemctl.quiet_run(*systemctl_args, "disable", "--now", service.service_name)
             end
           elsif System.launchctl?
-            quiet_system System.launchctl, "bootout", "#{System.domain_target}/#{service.service_name}"
-            unless no_wait
-              time_slept = 0
-              sleep_time = 1
-              max_wait = T.must(max_wait)
-              while ($CHILD_STATUS.to_i == 9216 || service.loaded?) && (max_wait.zero? || time_slept < max_wait)
-                sleep(sleep_time)
-                time_slept += sleep_time
-                quiet_system System.launchctl, "bootout", "#{System.domain_target}/#{service.service_name}"
+            domain_targets = System.candidate_domain_targets
+            dont_wait_statuses = [3, 125] # 3 = no such process, 125 = invalid domain
+            while service.loaded? && !domain_targets.empty?
+              domain_target = domain_targets.shift
+              quiet_system System.launchctl, "bootout", "#{domain_target}/#{service.service_name}"
+              unless no_wait
+                time_slept = 0
```

Comment:
Feels like `no_wait` would ideally be a e.g. for loop parameter to ensure there's only a single iteration

Reply:
We probably still want an iterations per target for `no_wait` as they are not the same operation. `no_wait` prevents repeatedly calling the same `bootout` though.